addons/sale_loyalty_delivery/models/sale_order.py

```diff
@@ -4,15 +4,26 @@
 from odoo import _, models
 from odoo.fields import Command
 
+
 class SaleOrder(models.Model):
     _inherit = 'sale.order'
 
+    # delivery overrides
+
+    def _compute_amount_total_without_delivery(self):
+        res = super()._compute_amount_total_without_delivery()
+        return res - sum(
+            self.order_line.filtered(
+                lambda l: l.coupon_id and l.coupon_id.program_type in ['ewallet', 'gift_card']
+            ).mapped('price_unit')
+        )
+
+    # sale_loyalty overrides
+
     def _get_no_effect_on_threshold_lines(self):
-        self.ensure_one()
```

Comment:
I don't mind that you remove this from overrides, but please apply it to all overrides of this file then (`_get_reward_line_values`)

Reply:
Only removing it when the super call does the check and can be called first, which isn't the case in `_get_reward_line_values`